src/Components/TypingTest.js

```diff
@@ -101,9 +107,31 @@ class TypingTest extends Component {
     }, 4000);
   }
 
+  checkKey(value) {
+    if (this.wordsTest.started) {
+      const charCheck = this.wordsTest.checkKeyChar(value);
+      if (charCheck.isCharCorrect === true) {
+        this.setDisplayText(false);
+      } else {
+        this.setDisplayText(false, charCheck.errorText);
+      }
+    }
+  }
+
   finishedFunction() {
     this.wordsTest.started = false;
-    this.setState({ displayString: 'You have finished' });
```

Comment:
ad the line 

`
this.wordsTest.finishStopWatch();
`

on the finished function

Reply:
I have merged your pull request to wpm test. I was looking at your comments out of order and I didn't really look at your pull request for wpm test. My bad.

Reply:
No problem @Scoombe , I've actually made another PR on wpm test as the TS needed building, that then should allow this PR to work :)

Reply:
https://github.com/Scoombe/wpmTest/pull/30